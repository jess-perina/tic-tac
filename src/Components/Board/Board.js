import React, { Component } from 'react';
import './Board.css';
import Square from '../Square/Square';

export default class Board extends Component {
  renderSquare(i) {
    return (
      <Square
        key={i}
        value={this.props.squares[i]}
        onClick={() => this.props.onClick(i)}
      />
    );
  }

  renderBoard() {
    let board = [];
    for (let i = 0; i < 3; i++) {
      let squares = [];
      for (let j = 0; j < 3; j++) {
        squares.push(this.renderSquare(i + j + (i * 2)))
      }
      board.push(<div className="board-row" key={i}>{squares}</div>)
    }
    return board
  }

  render() {
    return (
      <div className="game-container">
        {this.renderBoard()}
      </div>
    );
  }
}
