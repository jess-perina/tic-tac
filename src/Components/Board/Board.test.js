import React from 'react'
import { shallow } from 'enzyme'
import Board from './Board'

describe('Board Component', () => {
  const mockSquares = []
  it('renders as espected', () => {
    expect(shallow(<Board squares={mockSquares}/>)).toMatchSnapshot()
  })
})