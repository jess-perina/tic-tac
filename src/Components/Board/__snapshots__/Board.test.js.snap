// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board Component renders as espected 1`] = `ShallowWrapper {}`;
